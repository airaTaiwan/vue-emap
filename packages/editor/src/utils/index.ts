import type { Point } from '@airataiwan/utils'

import { CursorMap } from '../types/control'

export function setCornerCursor(corner: Point, center: Point): CursorMap {
  if (corner.y < center.y) {
    if (corner.x < center.x)
      return CursorMap.TL
    else if (corner.x > center.x)
      return CursorMap.TR
    else return CursorMap.MT
  }
  else if (corner.y > center.y) {
    if (corner.x < center.x)
      return CursorMap.BL
    else if (corner.x > center.x)
      return CursorMap.BR
    else return CursorMap.MB
  }
  else {
    if (corner.x < center.x)
      return CursorMap.ML
    else return CursorMap.MR
  }
}
