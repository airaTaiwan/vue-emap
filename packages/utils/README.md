# @vue-emap/utils
`Vue-Emap` 的 `utils` 工具集，提供了一些常用的工具函数。

## Usage

```bash
pnpm i @vue-emap/utils
```

## License

[MIT](../../LICENSE) License © 2024 [airaTaiwan](https://github.com/airaTaiwan456)
