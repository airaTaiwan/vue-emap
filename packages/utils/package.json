{
  "name": "@airataiwan/utils",
  "version": "0.0.1-beta.3",
  "packageManager": "pnpm@9.5.0",
  "description": "Vue EMap utils library.",
  "author": "airaTaiwan <https://github.com/airaTaiwan456>",
  "license": "MIT",
  "homepage": "https://www.aira.com.tw/",
  "repository": {
    "directory": "packages/utils",
    "type": "git",
    "url": "git+https://github.com/airaTaiwan/vue-emap.git"
  },
  "bugs": {
    "url": "https://github.com/airaTaiwan/vue-emap/issues"
  },
  "keywords": [
    "vue",
    "map",
    "utils"
  ],
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.mjs",
      "require": "./dist/index.js"
    }
  },
  "main": "src/index.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "tsup",
    "watch": "tsup --watch"
  },
  "peerDependencies": {
    "vue": "^3.3.0"
  },
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/"
  }
}
