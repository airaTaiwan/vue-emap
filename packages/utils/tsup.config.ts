import { defineConfig } from 'tsup'

export default defineConfig({
  clean: true,
  dts: true,
  entry: ['src/index.ts'],
  external: ['vue'],
  format: ['cjs', 'esm'],
  minify: true,
})
