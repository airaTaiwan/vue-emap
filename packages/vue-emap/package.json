{
  "name": "vue-emap",
  "type": "module",
  "version": "0.0.0",
  "packageManager": "pnpm@9.4.0",
  "description": "Vue EMap component library.",
  "author": "airaTaiwan <https://github.com/airaTaiwan456>",
  "license": "MIT",
  "homepage": "https://www.aira.com.tw/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/airaTaiwan/vue-emap.git",
    "directory": "packages/vue-emap"
  },
  "bugs": {
    "url": "https://github.com/airaTaiwan/vue-emap/issues"
  },
  "keywords": [
    "vue",
    "map",
    "vue-emap"
  ],
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/src/index.d.ts",
      "import": "./dist/index.js",
      "require": "./dist/index.umd.cjs"
    },
    "./dist/index.css": "./dist/index.css"
  },
  "main": "./dist/index.umd.cjs",
  "module": "./dist/index.js",
  "types": "./dist/src/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "pnpm type-check && pnpm build-only",
    "build-only": "vite build",
    "watch": "vite build --watch",
    "type-check": "vue-tsc -p tsconfig.json --noEmit",
    "type-gen": "vue-tsc --declaration  --emitDeclarationOnly",
    "test": "vitest",
    "pub:release": "pnpm publish --access public"
  },
  "dependencies": {
    "@vue-emap/core": "workspace:*",
    "@vue-emap/utils": "workspace:*"
  }
}
