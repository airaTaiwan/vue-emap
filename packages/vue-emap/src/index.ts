export * from '@airataiwan/core'
// styles
export * from '@airataiwan/core/dist/index.css'

export * from '@airataiwan/utils'
