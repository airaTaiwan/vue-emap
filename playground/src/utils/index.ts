import type { Point } from 'vue-emap'

export function generateMarkerPos(rangeX: number, rangeY: number): Point {
  const x = Math.floor(Math.random() * rangeX) + 1
  const y = Math.floor(Math.random() * rangeY) + 1

  return {
    x,
    y,
  }
}
