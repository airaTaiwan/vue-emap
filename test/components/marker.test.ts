import { sleep } from '@antfu/utils'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import { h } from 'vue'
import { EMap, Marker } from 'vue-emap'

import mapImg from './mock/map.jpg'

describe('map', () => {
  it('marker render test', () => {
    const wrapper = mount(EMap, {
      props: {
        img: mapImg,
        zoomControl: true,
      },
      slots: {
        default: Marker,
      },
    })

    expect(wrapper.find('.marker').exists()).toBeTruthy()
  })

  it('zoom to 1.5', async () => {
    const wrapper = mount(EMap, {
      props: {
        img: mapImg,
        zoomControl: true,
      },
      slots: {
        default: Marker,
      },
    })

    expect(wrapper.find('.marker').html()).toMatchSnapshot()

    wrapper.vm.setZoom(1.5)

    await sleep(450)

    expect(wrapper.find('.marker').html()).toMatchSnapshot()
  })
})
