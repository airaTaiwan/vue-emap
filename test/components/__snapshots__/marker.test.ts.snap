// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`map > zoom to 1.5 1`] = `"<div style="top: 300.00px; left: 500.00px;" class="marker" block="" op100="" pos-absolute="" z0=""> I am a marker </div>"`;

exports[`map > zoom to 1.5 2`] = `"<div style="top: 449.88px; left: 749.79px;" class="marker" block="" op100="" pos-absolute="" z0=""> I am a marker </div>"`;
